### PR TITLE
[MNTR-7] Configure CI to handle automatic deploy to Heroku

### LINKED ISSUE #7: Configure CI to handle automatic deploy to Heroku
Deploy every commit on the master branch to Heroku automatically

### PR DESCRIPTION
**Pull request by sajti12**
Resolves #7

### CODE REVIEW

**Comment by sajti12** [comment]
Please note that from now on two environment variables are required to be set in CircleCI:
- `HEROKU_API_KEY`: required by Heroku for authentication
- `HEROKU_APP_NAME`: supplied to deploy command

File: .circleci/config.yml [hunk 1]
```diff
@@ -17,10 +20,14 @@ jobs:
              - ~/.m2
            key: mentoring-{{ checksum "pom.xml" }}
        - store_artifacts:
-           path: target/WAR_FILE_NAME
+           path: target/mentoring-0.0.1-SNAPSHOT.war
+           destination: mentoring-0.0.1-SNAPSHOT.war
```
Comment (pwdypb) [inline .circleci/config.yml:24]: Should use `$WAR_FILE_NAME` variable

File: .circleci/config.yml [hunk 2]
```diff
@@ -17,10 +20,14 @@ jobs:
...
+       - persist_to_workspace:
+           root: target
+           paths:
+              - mentoring-0.0.1-SNAPSHOT.war
```
Comment (pwdypb) [inline .circleci/config.yml:28]: Same here

**Review by pwdypb — CHANGES_REQUESTED** [review]

**Comment by pwdypb** [comment]
The app will be available on: https://trexxx.herokuapp.com/

File: .circleci/config.yml (hunk 1 above)
Comment (sajti12) [reply .circleci/config.yml:24]: It isn't supported at the moment. Refer to [this](https://discuss.circleci.com/t/using-environment-variables-in-config-yml-not-working/14237).

File: .circleci/config.yml (hunk 2 above)
Comment (sajti12) [reply .circleci/config.yml:28]: See above

**Review by pwdypb — APPROVED** [review]

**Pull request merged**